### PR TITLE
 Skip network creation if network already exists in HNS

### PR DESCRIPTION
**Pull request by aegal**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: network/hnswrapper/hnsv2wrapperfake.go [hunk 1]
```diff
@@ -167,7 +167,7 @@ func (f Hnsv2wrapperFake) GetNetworkByName(networkName string) (*hcn.HostCompute
 	if network, ok := f.Cache.networks[networkName]; ok {
 		return network.GetHCNObj(), nil
 	}
-	return &hcn.HostComputeNetwork{}, nil
+	return nil, hcn.NetworkNotFoundError{}
```
Comment (tamilmani1989) [inline network/hnswrapper/hnsv2wrapperfake.go:170]: just make sure cni code doesn't dereference pointer without checking for error

File: network/network_windows.go [hunk 2]
```diff
@@ -310,14 +311,25 @@ func (nm *networkManager) newNetworkImplHnsV2(nwInfo *NetworkInfo, extIf *extern
 		return nil, err
 	}
 
-	// Create the HNS network.
-	log.Printf("[net] Creating hcn network: %+v", hcnNetwork)
-	hnsResponse, err := hnsv2.CreateNetwork(hcnNetwork)
+	// check if network exists, only create the network does not exist
+	hnsResponse, err := hnsv2.GetNetworkByName(hcnNetwork.Name)
```
Comment (tamilmani1989) [inline network/network_windows.go:315]: can you also log if network exists

File: network/network_windows.go [hunk 3]
```diff
@@ -310,14 +311,25 @@ func (nm *networkManager) newNetworkImplHnsV2(nwInfo *NetworkInfo, extIf *extern
...
+
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create hcn network: %s due to error: %v", hcnNetwork.Name, err)
-	}
+		// if network not found, create the HNS network.
+		if errors.As(err, &hcn.NetworkNotFoundError{}) {
```
Comment (tamilmani1989) [inline network/network_windows.go:319]: can you add ut for testing these cases?

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: network/network_windows.go (hunk 3 above)
Comment (aegal) [reply network/network_windows.go:319]: added

File: network/network_windows.go (hunk 2 above)
Comment (aegal) [reply network/network_windows.go:315]: sure, added it when err is empty

File: network/hnswrapper/hnsv2wrapperfake.go (hunk 1 above)
Comment (aegal) [reply network/hnswrapper/hnsv2wrapperfake.go:170]: this is the hns mock wrapper and only used by test code, know real cni code uses this type

**Review by tamilmani1989 — APPROVED** [review]
lgtm 🚢

**Pull request merged**